include/tags.h

```diff
@@ -56,6 +72,14 @@ struct lz_tag_hash {
 	u8 digest[];
 } __packed;
 
+struct lz_tag_setup_indirect {
+	struct lz_tag_hdr hdr;
+	/* type = SETUP_INDIRECT */
```

Comment:
Is it the case that both of these structures are used at the same time? If not, could these be put under a union?

Reply:
that is how they are used in the kernel. one is nested in the other

Reply:
Maybe make it nested in here as well? We probably won't be using (read: don't have enough space in SLB) any other setup_data.

Reply:
Sorry I don't understand the last comment. It is nested, that is how setup_indirect works. The structures are sequential in the final lz_tag_setup_indirect struct but the effect is the same.

Reply:
Yes, in memory it is the same, this is just a cosmetic change. What I had in mind is something like:
```
struct setup_data {
	u64 next;
	u32 type;
	u32 len;
	struct setup_indirect indirect;
}
```

Reply:
If you want sure.

Reply:
Done